### ISSUE TITLE
Filter bad/Offensive words from a list

### ISSUE THREAD

**Issue by meteorplus** [RocketChat/Rocket.Chat#1283]
Hi

I think it's important to have since users can use bad/offensive words it's easy we can have a very good words list from either www.noswearing.com or www.bannedwordlist.com and replace(badword,'****') with stars :)

I can do this ASAP I just need to know where to start thanks :)
##

**Comment by MyPod** [comment]
censoring should probably be an option selectable by the users, with an admin setting for auto-enabling, perhaps. Letting people add custom words would be too something to do - and, in parallel, allow to set words that "ping" users

**Comment by capensisma** [comment]
adding it as an option is easy :)

I have found a good nodejs [package](https://www.npmjs.com/package/bad-words)

we use right now and have a some code for it:

```javascript
RocketChat.callbacks.add(beforeSaveMessage, function,(message){
return filter(message.msg);// filter is a function from that package!
},1);
```

but where can add this ? and how do I deploy it?

Thanks
@MyPod

**Comment by capensisma** [comment]
I have got everything here code and the way to add it but it needs just to be added as an option in the admin panel so user can enable/disable it and/or set a banned list of words in textarea...

It's pretty easy to do but I don't know where should I add it??

**Comment by djeber** [comment]
any update on this one !!

**Comment by mchilson** [comment]
Yes, I think this is a viable feature also that needs to be included in any chat.

**Comment by xj721386** [comment]
I am just wondering if capensisma have figured out where to actually do the filtering?

**Comment by matthijs2704** [comment]
@capensisma have you figured out yet how to do this?

**Comment by capensisma** [comment]
@matthijs2704 @xj721386 sorry I was sleeping  :D do you want me to do a PR right now for this? I have many contribution on hold lol just lazy :'(

**Comment by matthijs2704** [comment]
whoops did I wake you? But if you can do a PR, yes please! :smile:

**Comment by capensisma** [comment]
@matthijs2704 ok I am Officially awake now :D Thanks 

Let's do this! :+1:

**Comment by capensisma** [comment]
@matthijs2704 better add a bounty to this and I will send a PR

**Comment by capensisma** [comment]
It's working :D OMG 
lemme send a PR after I clean up the code a little lol :D

**Comment by capensisma** [comment]
@matthijs2704 @xj721386 @mchilson @djeber @MyPod  **It's DONE**
https://github.com/RocketChat/Rocket.Chat/pull/2940

**Comment by psybernoid** [comment]
Great job on adding this in, but one problem - I notice it adds a default list of words. Is there any way to flush this ? I've attempted to edit the lang.js file and restarting the server, but entries in that file I've removed still get filtered out.

**Comment by MartinSchoeler** [comment]
Closing this due #2940
